### PR TITLE
step definition supports in php file

### PR DESCRIPTION
**Pull request by shth**
### 🤔 What's changed?

Added queries for tree sitter to grep the step definition node.

### ⚡️ What's your motivation? 

I want to have code completion in php files

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### ♻️ Anything particular you want feedback on?

First of all, thanks to the person reviewing this :bow:. I am new to cucumber and tree-sitter and new to contributing to open source in general. So forgive me if something is obviously wrong.

Back to the code:
I could not find any docs about defining parameter types in PHP. Can anyone give me some pointers to docs/guides on how to do it? Thanks!

### 📋 Checklist:

- [x] finish parameter types query
- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
Hi @shth - thanks for contributing this! I'm going to be away for little over a week - hope to look into this when I'm back.

**Comment by shth** [comment]
@aslakhellesoy No problem! Thanks in advance for the help

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "build": "npm run build:cjs && npm run build:esm",
     "build:java-wasm": "npx tree-sitter build-wasm node_modules/tree-sitter-java",
     "build:typescript-wasm": "npx tree-sitter build-wasm node_modules/tree-sitter-typescript/typescript",
+    "build:php-wasm": "npx tree-sitter build-wasm node_modules/tree-sitter-php",
```
Comment (aslakhellesoy) [inline package.json:27]: Remove this

File: package.json [hunk 2]
```diff
@@ -66,8 +67,9 @@
     "npm-check-updates": "12.5.7",
     "prettier": "2.6.2",
     "pretty-quick": "3.1.3",
-    "txtgen": "3.0.1",
+    "tree-sitter-php": "^0.19.0",
```
Comment (aslakhellesoy) [inline package.json:70]: Remove this

File: src/tree-sitter/helpers.ts [hunk 3]
```diff
@@ -17,7 +17,9 @@ export function makeParameterType(name: string, regexp: string | RegExp) {
 }
 
 export function toStringOrRegExp(s: string): string | RegExp {
-  const match = s.match(/^([/'"])(.*)([/'"])$/)
+  let match = s.match(/^([/'"])(.*)([/'"])$/)
+  // match multi-line comments
```
Comment (aslakhellesoy) [inline src/tree-sitter/helpers.ts:21]: As we add more languages, I think this kind of branching can get a bit messy. No need to fix in this PR, just adding a note as a reminder to refactor this.

**Review by aslakhellesoy — CHANGES_REQUESTED** [review]
This looks like a good start. I haven't had a chance to run the code yet. Could you please update your PR with the `main` branch? The changes from #30 will introduce some merge conflicts, but overall it should simplify the code a bit.

**Comment by aslakhellesoy** [comment]
I've made some more changes on `main` to make it easier to add support for a new language. Please see [CONTRIBUTING.md](https://github.com/cucumber/language-service/blob/main/CONTRIBUTING.md)

**Comment by shth** [comment]
Hi @aslakhellesoy, I have rebased to the `main` branch. Please have a look.

Sorry I went silent for a few days.

File: test/tree-sitter/testdata/php/StepDefinitions.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace LanguageService;
+
+class StepDefinitions 
+{
+    /**
+     * @Given ^a regexp$
```
Comment (shth) [inline test/tree-sitter/testdata/php/StepDefinitions.php:8]: May I know why we are matching for `^` and `$` literally in `ExpressionBuilder.test.ts`?

**Comment by shth** [comment]
Also thanks for the refactoring! It feels much cleaner to add languages now.

File: test/tree-sitter/testdata/php/StepDefinitions.php (hunk 4 above)
Comment (aslakhellesoy) [reply test/tree-sitter/testdata/php/StepDefinitions.php:8]: This is to make sure the entire regexp is extracted from code.

**Review by aslakhellesoy — COMMENTED** [review]
Excellent work @shth! Regarding parameter types - this is a [Cucumber Expressions](https://github.com/cucumber/cucumber-expressions#readme) concept. Since Cucumber Expressions aren't yet ported to PHP and integrated into Behat, it's correct to leave the `defineParameterTypeQueries` as an empty array.

/cc @ciaranmcnulty

**Comment by shth** [comment]
@aslakhellesoy thanks for the explanation. I have fixed the eslint errors btw.

**Comment by aslakhellesoy** [comment]
Hi @shth,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**